### PR TITLE
fix:  `pixi add` with more than just package name and version

### LINKED ISSUE #1697: Tensorflow GPU install
### Problem description

Hi, in the docs of tensorflow they advice to use:
```sh
python3 -m pip install tensorflow[and-cuda]
```
Any idea how to reproduce this with pixi ?

### PR DESCRIPTION
**Pull request by ruben-arts**
Fixes #1697 
We broke `pixi add --pypi pytest[dev]` as it didn't add the [dev] anymore. Now that is fixed.

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]

**Comment by ruben-arts** [comment]
@baszalmstra I wasn't happy, and found out `pixi add pytorch::pytorch` was also broken. I now also fixed that in this pr and just reuse the requirement and matchspec as is.

File: tests/add_tests.rs [hunk 1]
```diff
@@ -75,6 +77,29 @@ async fn add_functionality() {
     ));
 }
 
+/// Test adding a package with a specific channel
+#[tokio::test]
+async fn add_with_channel(){
+    let pixi = PixiControl::new().unwrap();
+
+    pixi.init()
+        .no_fast_prefix_overwrite(true)
```
Comment (ruben-arts) [inline tests/add_tests.rs:86]: Can't use url's in matchspecs yet, fix in: https://github.com/conda-incubator/rattler/pull/792

File: tests/add_tests.rs [hunk 2]
```diff
@@ -75,6 +77,29 @@ async fn add_functionality() {
...
+        .await
+        .unwrap();
+
+    pixi.add("conda-forge::py_rattler").without_lockfile_update().await.unwrap();
```
Comment (ruben-arts) [inline tests/add_tests.rs:90]: Currently still a slow test as there is a bug in this `--no-lockfile-update` so it is actually solving. This is fixed in: https://github.com/prefix-dev/pixi/pull/1683

Comment (ruben-arts) [reply tests/add_tests.rs:90]: Otherwise I would have used the mock PackageDatabase :wink:

**Review by ruben-arts — COMMENTED** [review]
Some mentions of why I added some test stuff.

**Review by tdejager — APPROVED** [review]

**Pull request merged**